### ISSUE TITLE
Regression: args[module] Unsupported parameters

### ISSUE THREAD

**Issue by nre-ableton** [ansible/ansible-lint#3138]
##### Summary

In ansible-lint 6.10.0, some modules started to raise `args[module]` violations (see https://github.com/ansible/ansible-lint/issues/2830). This was fixed and shipped in 6.13.1 but appears to be broken again in 6.14.0.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.14.0 using ansible 2.14.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Example play (see [source here](https://github.com/Ableton/ansible-role-jenkins-jcasc/blob/main/tasks/start-service.yml)):

```yaml
- name: Start Jenkins service
  become: true
  ansible.builtin.service:
    name: "jenkins"
    state: started
    # This normally isn't required, but we also test this role in Docker containers that
    # resemble standalone hosts with SSH. Since OpenSSH has libsystemd as a dependency,
    # it will fool the service module into selecting systemd as the service backend.
    use: "{{ ansible_service_mgr }}"
  # This is required or else molecule's idempotency test will fail. See:
  # https://github.com/ansible-community/molecule/issues/2765
  changed_when: false
  when: not ansible_check_mode
```

ansible-lint 6.14.0 fails on this play with this error:

> `tasks/start-service.yml:2: args[module]: Unsupported parameters for (basic.py) module: use. Supported parameters include: arguments, enabled, name, pattern, runlevel, sleep, state (args). (warning)`

**Comment by ssbarnea** [comment]
@ganeshrn ^ The way args tries to validate modules does not seem be very reliable as lots of modules do dark magic about their arguments. They document something but internally they do modify arguments before passing them and generating problems like this one.

I wonder how long our workarounds lists are going to grow https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/rules/args.py#L41-L52

**Comment by MarkusTeufelberger** [comment]
I wonder when this workaround list will start to have to be compared against `ansible-core` version numbers. :-D

**Comment by nre-ableton** [comment]
@ssbarnea Thanks for the quick fix!
